### PR TITLE
Add CreateContainerError and InvalidImageName waiting reasons

### PR DESCRIPTION
**Pull request by jdewald**
**What this PR does / why we need it**:

Adds 'CreateContainerError' and 'InvalidImageName' to the available container waiting reasons, which we need visibility into in our production environment

### CODE REVIEW

File: internal/collector/pod.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ const nodeUnreachablePodReason = "NodeLost"
 
 var (
 	descPodLabelsDefaultLabels = []string{"namespace", "pod"}
-	containerWaitingReasons    = []string{"ContainerCreating", "CrashLoopBackOff", "CreateContainerConfigError", "ErrImagePull", "ImagePullBackOff"}
+	containerWaitingReasons    = []string{"ContainerCreating", "CrashLoopBackOff", "CreateContainerConfigError", "ErrImagePull", "ImagePullBackOff","CreateContainerError","InvalidImageName"}
```
Comment (tariq1890) [inline internal/collector/pod.go]: there needs to be spaces after the commas.
```suggestion
	containerWaitingReasons    = []string{"ContainerCreating", "CrashLoopBackOff", "CreateContainerConfigError", "ErrImagePull", "ImagePullBackOff", "CreateContainerError", "InvalidImageName"}
```

**Comment by tariq1890** [comment]
@jdewald Welcome and Thank you very much for contributing :). I think the failing unit tests need to be taken care of.

PR looks good otherwise.

**Comment by tariq1890** [comment]
Also, please don't forget to squash your commit history after making the required changes.

**Comment by jdewald** [comment]
@tariq1890 Thanks! I totally should have run the linting before I committed!

**Comment by jdewald** [comment]
Applying squash now

**Comment by jdewald** [comment]
It looks like test is failing, though it passed locally. Seeing if I can reproduce.

**Pull request merged**